packages/integration-tests/tests/username-password-flow.test.ts

```diff
@@ -0,0 +1,203 @@
+    expect(response.body).toBe('Redirecting to <a href="/sign-in">/sign-in</a>.');
+
+    const cookie = extractCookie(response);
+    expect(cookie).toBeTruthy();
+    updateCookie(cookie);
+  });
+
+  it('should register with username and password and redirect to oidc/auth endpoint to start an auth process', async () => {
+    type RegisterResponse = {
+      redirectTo: string;
+    };
+
+    const registerResponse = await api
+      .post('session/register/username-password', {
+        headers: {
+          cookie: getInteractionCookie(),
+        },
+        json: account,
+      })
+      .json<RegisterResponse>();
+
+    const { redirectTo } = registerResponse;
+
+    expect(redirectTo).toBeTruthy();
+    expect(redirectTo.startsWith(`${logtoUrl}/oidc/auth`)).toBeTruthy();
+  });
+
+  it('should sign in with username and password and redirect to oidc/auth endpoint to start an auth process', async () => {
+    type SignInResponse = {
+      redirectTo: string;
+    };
+
+    const signInResponse = await api
+      .post('session/sign-in/username-password', {
+        headers: {
+          cookie: getInteractionCookie(),
+        },
+        json: account,
+        followRedirect: false,
+      })
+      .json<SignInResponse>();
+
+    const { redirectTo: invokeAuthUrl } = signInResponse;
+
+    expect(invokeAuthUrl).toBeTruthy();
+    expect(invokeAuthUrl.startsWith(`${logtoUrl}/oidc/auth`)).toBeTruthy();
+
+    setNextRedirectTo(invokeAuthUrl);
+  });
+
+  it('should invoke the auth process and redirect to the consent page with session cookie', async () => {
+    const invokeAuthUrl = getNextRedirectTo();
+    const invokeAuthResponse = await got.get(invokeAuthUrl, {
+      headers: {
+        cookie: getInteractionCookie(),
+      },
+      followRedirect: false,
+    });
+
+    // Note: Redirect to consent page
+    expect(invokeAuthResponse).toHaveProperty('statusCode', 303);
+    expect(invokeAuthResponse.body).toBe(
+      'Redirecting to <a href="/sign-in/consent">/sign-in/consent</a>.'
+    );
+
+    const cookie = extractCookie(invokeAuthResponse);
+    expect(cookie).toBeTruthy();
+    expect(cookie.includes('_session.sig')).toBeTruthy();
+
+    updateCookie(cookie);
+  });
+
+  it('should redirect to oidc/auth endpoint to complete the auth process after consent', async () => {
+    type ConsentResponse = {
+      redirectTo: string;
+    };
+
+    const consentResponse = await api
+      .post('session/consent', {
+        headers: {
+          cookie: getInteractionCookie(),
+        },
+        followRedirect: false,
+      })
+      .json<ConsentResponse>();
+
+    const { redirectTo: completeAuthUrl } = consentResponse;
+
+    expect(completeAuthUrl).toBeTruthy();
+    expect(completeAuthUrl.startsWith(`${logtoUrl}/oidc/auth`)).toBeTruthy();
+
+    setNextRedirectTo(completeAuthUrl);
+  });
+
+  it('should get the authorization code from the callback uri when the auth process is completed', async () => {
```

Comment:
seems that: the authorization process should be completed after fetching tokens by authorization code?

Reply:
No, the authorization process returns the authorization code.
the fetch token step comes after the process is completed.